internal/services/oracledatabase/cloud_vm_cluster_resource.go

```diff
@@ -0,0 +1,429 @@
+	ClusterName              string                       `tfschema:"cluster_name"`
+	DataCollectionOptions    []DataCollectionOptionsModel `tfschema:"data_collection_options"`
+	DataStoragePercentage    int64                        `tfschema:"data_storage_percentage"`
+	IsLocalBackupEnabled     bool                         `tfschema:"is_local_backup_enabled"`
+	IsSparseDiskgroupEnabled bool                         `tfschema:"is_sparse_diskgroup_enabled"`
+	TimeZone                 string                       `tfschema:"time_zone"`
+}
+
+func (CloudVmClusterResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		// Azure
+		"location": commonschema.Location(),
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+		"resource_group_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+		"tags": commonschema.Tags(),
+
+		// Required
+		"cloud_exadata_infrastructure_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		"cpu_core_count": {
+			Type:     pluginsdk.TypeInt,
+			Required: true,
+		},
+		"data_storage_size_in_tbs": {
+			Type:     pluginsdk.TypeFloat,
+			Optional: true,
+			Computed: true,
+			ForceNew: true,
+		},
+		"db_node_storage_size_in_gbs": {
+			Type:     pluginsdk.TypeInt,
+			Optional: true,
+			Computed: true,
+		},
+		"db_servers": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+		"display_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+		"gi_version": {
+			Type:             pluginsdk.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			DiffSuppressFunc: GiVersionDiffSuppress,
+		},
+		"hostname": {
+			Type:             pluginsdk.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			DiffSuppressFunc: DbSystemHostnameDiffSuppress,
+		},
+		"license_model": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+		"memory_size_in_gbs": {
+			Type:     pluginsdk.TypeInt,
+			Optional: true,
+			Computed: true,
+		},
+		"ssh_public_keys": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+		"subnet_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		"vnet_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		// Optional
+		"backup_subnet_cidr": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"cluster_name": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Computed: true,
+			ForceNew: true,
+		},
+		"data_collection_options": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			MaxItems: 1,
+			MinItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"is_diagnostics_events_enabled": {
```

Comment:
i think we can drop is?
```suggestion
					"diagnostics_events_enabled": {
```